src/internal/operators/tap.ts

```diff
@@ -78,6 +78,12 @@ class DoOperator<T> implements Operator<T, T> {
               private complete?: () => void) {
   }
   call(subscriber: Subscriber<T>, source: any): TeardownLogic {
+    /* make sure throwIfEmpty hasValue is set to false on re-subscribe */
+    if (this.nextOrObserver) {
+      if (this.nextOrObserver['hasValue']) {
```

Comment:
We can't do this here. Anyone that passes any object with a `hasValue` on it will have that mutated. It's too unpredictable.

More likely, we have a bug in the `throwIfEmpty` implementation that we need to clear up.